test/e2e_node/runtime_conformance_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	retryTimeout = time.Minute * 4
+	pollInterval = time.Second * 5
+)
+
+var _ = Describe("Container Runtime Conformance Test", func() {
+	var cl *client.Client
+
+	BeforeEach(func() {
+		// Setup the apiserver client
+		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
+	})
+
+	Describe("container conformance blackbox test", func() {
+		Context("when running a container that terminates", func() {
+			var containerCase ConformanceContainer
+
+			BeforeEach(func() {
+				containerCase = ConformanceContainer{
+					Container: api.Container{
+						Image:           "gcr.io/google_containers/busybox",
+						Name:            "busybox",
+						Command:         []string{"sh", "-c", "env"},
+						ImagePullPolicy: api.PullIfNotPresent,
+					},
+					Client:   cl,
+					Phase:    api.PodSucceeded,
+					NodeName: *nodeName,
+				}
+				err := containerCase.Create()
+				Expect(err).NotTo(HaveOccurred())
+
+				//Waiting for the 'running' phase
+				for start := time.Now(); time.Since(start) < retryTimeout; time.Sleep(pollInterval) {
```

Comment:
Eventually will popup an 'error' that is something we want to tell in our test.
I don't want to see this kind of error in the 'BeforeEach' session.

In the 'BeforeEach' session, I just want a pod to run for a while or pass the 'pending' phase.
Which means the status is changed. It is ok If it becomes 'terminated' or 'succeed'.